pyaedt/modules/solutions.py

```diff
@@ -1071,338 +1075,248 @@ def ifft_to_file(
-        CenterB = 0
-        CenterX = 0
-        CenterY = 0
-
-        # collecting all active cells inside the specified region
-        activeCells = []
-
-        for i in range(0, len(self.all_port_names)):
-            index_str = self.get_array_index(self.all_port_names[i])
-            row = index_str[1]
-            col = index_str[0]
-            a = row
-            b = col
-
-            activeCells.append((a, b))  # because ffd is assuming all ffd files are active
-        if len(activeCells) == 0:
-            return
-
-        [a_min, a_max, b_min, b_max] = self.array_min_max_values()
-
-        CenterA = (a_min + a_max) / 2
-        CenterB = (b_min + b_max) / 2
-        CenterX = (CenterA + 0.5) * self.Ax + (CenterB + 0.5) * self.Bx
-        CenterY = (CenterA + 0.5) * self.Ay + (CenterB + 0.5) * self.By
-
-        self.CenterA = CenterA
-        self.CenterB = CenterB
-        self.CenterX = CenterX
-        self.CenterY = CenterY
-        # find the distance from the edge to the center
-        AMax = a_max - a_min
-        BMax = b_max - b_min
-
-        self.AMax = AMax
-        self.BMax = BMax
-        for a, b in activeCells:
-            x = (a + 0.5) * self.Ax + (b + 0.5) * self.Bx
-            y = (a + 0.5) * self.Ay + (b + 0.5) * self.By
-            x_dis = abs(x - CenterX)
-            y_dis = abs(y - CenterY)
-            distance = math.sqrt(x_dis**2 + y_dis**2)
-            XMax = max(XMax, x_dis)
-            YMax = max(YMax, y_dis)
-            RMax = max(RMax, distance)
-
-        self.RMax = RMax
-        self.XMax = XMax
-        self.YMax = YMax
-        self.RMax *= 2
-        self.XMax *= 2
-        self.YMax *= 2
-        return True
-
-    @pyaedt_function_handler()
-    def element_location(self, a, b):
-        """Element location in the array.
-
-        Parameters
-        ----------
-        a : int
-        b : int
+            self.farfield_data = self.combine_farfield()
 
-        Returns
-        -------
-        list of float
+    @property
+    def taper(self):
+        """Taper type.
+
+        Options are:
+        - "flat"
+        - "uniform"
+        - "cosine"
+        - "triangular"
+        - "hamming"
         """
-        a = int(a)
-        b = int(b)
+        return self._taper
 
-        x = (a + 0.5) * self.Ax + (b + 0.5) * self.Bx
-        y = (a + 0.5) * self.Ay + (b + 0.5) * self.By
-        x_dis = x - self.CenterX
-        y_dis = y - self.CenterY
+    @taper.setter
+    def taper(self, val):
+        if val.lower() in ("flat", "uniform", "cosine", "triangular", "hamming"):
+            self._taper = val
+        else:
+            self.logger.error("This taper is not implemented")
 
-        return np.array([x_dis, y_dis, 0])
+    @property
+    def origin(self):
+        """Far field origin in meters."""
+        return self._origin
+
+    @origin.setter
+    def origin(self, vals):
+        if len(vals) != 3:
+            self.logger.error("Wrong origin")
```

Comment:
```suggestion
            self.logger.error("Origin is wrong.")
```
Message strings should be complete sentences.